### PR TITLE
feat: Resizable dataset and controls panels on Explore view

### LINKED ISSUE #115: [explore] resizable data and control panel
Timeline: implement this feature along with drag and drop control panel 
Purpose: to display complete super long query values
likely keeping the current collapsible feature  

tbd....
- default width for both panels - currently 320px for both
- should we set min/max width when resizing

Designer:@mihir174

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
This PR enables resizing dataset and controls panels by dragging. Minimum width is 320px, maximum is 33% of screen width. The maximum value is set to ensure that the chart is still visible even when both panels are fully expanded - tested on 1440px screen width.
Closes https://github.com/apache-superset/superset-roadmap/issues/115

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
https://user-images.githubusercontent.com/15073128/104225586-def56480-5446-11eb-8376-bee2a74bb067.mov

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache-superset/superset-roadmap/issues/115
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

CC: @junlincc @villebro @ktmud
@adam-stasiak Can you help with testing?

### CODE REVIEW

**Comment by ktmud** [comment]
THANK YOU! This is long overdue. I will test and take a look at the code soon.

File: superset-frontend/src/explore/components/ExploreViewContainer.jsx
Comment (ktmud) [inline superset-frontend/src/explore/components/ExploreViewContainer.jsx:408]: I think the minWidth/default width for the Dataset column since it somehow looks wider visually comparing to the Controls column, probably because of the gray background or more white space.

**Review by ktmud — APPROVED** [review]
LGTM with some suggestions on column widths.

File: superset-frontend/src/explore/components/ExploreViewContainer.jsx
Comment (junlincc) [reply superset-frontend/src/explore/components/ExploreViewContainer.jsx:461]: 33% is already squeezing the viz. we tested

**Review by junlincc — APPROVED** [review]
ezgif-3-64d94cd98ab4

Manually tested! no noticeable issues found. work perfectly as provided in the instruction. 
Resizing will not trigger query or create performance issues

NOTE⚠️: 
- This is only Product and QA approval with NO code reviews 
- This approval indicates that all UI changes meet product requirements, no noticeable regression 
- This approval gives the green light to code review, thus saving the PMC's effort from reviewing the code repeatedly before the PR is ready
- Please do NOT merge this PR without at least one code review approval from PMC member
- Please do NOT merge PRs in(Explore and Dashboard) without Product and QA approval

**Comment by adam-stasiak** [comment]
Tested also - looks fine.
One question to @junlincc Should this icon change size while dragging?

https://user-images.githubusercontent.com/25153919/104239991-a449f700-545b-11eb-9ba2-2ae01dacb5cc.mov

**Comment by junlincc** [comment]
> Tested also - looks fine.
> One question to @junlincc Should this icon change size while dragging?
> 
>  Nagranie.z.ekranu.2021-01-11.o.22.22.55.mov



> Should this icon change size while dragging

really good find Adam!! dataset icon and ... icon both stretch a bit, but collapse icon doesnot. maybe because they are in different containers? we should address this in a separate PR to really reach a perfect state 😀

**Comment by junlincc** [comment]
good to merge

File: superset-frontend/src/explore/components/ExploreViewContainer.jsx
Comment (kgabryje) [reply superset-frontend/src/explore/components/ExploreViewContainer.jsx:461]: Increasing maxWidth wouldn't do much, as the chart section has a minWidth 512px set.

File: superset-frontend/src/explore/components/ExploreViewContainer.jsx
Comment (kgabryje) [reply superset-frontend/src/explore/components/ExploreViewContainer.jsx:408]: I think 300px for gray column looks ok. WDYT?
image

Comment (ktmud) [reply superset-frontend/src/explore/components/ExploreViewContainer.jsx]: Looking good! Thanks for the update!

**Comment by ktmud** [comment]
I think we can mark this as `v1.0.0` as well. @junlincc @rusackas  what do you think?

**Comment by junlincc** [comment]
> I think we can mark this with `v1.0.0` as well. @junlincc @rusackas @villebro what do you think?

that is the plan! 
@villebro pls make sure this one gets in.

**Pull request merged**
